.github/workflows/devcontainer.yaml

```diff
@@ -0,0 +1,63 @@
+name: devcontainer
+
+on:
+  push:
+    # ToDo: Add partial Taskfile (only include rust-tools)
+    paths:
+      - .devcontainer/base-image/Dockerfile
+  pull_request:
+    paths:
+      - .devcontainer/base-image/Dockerfile
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: docker/metadata-action@v4
+        id: meta
+        with:
+          images: ghcr.io/${{ github.repository_owner }}/prql-devcontainer-base
+          tags: |
+            type=raw,latest
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - id: set-platforms
+        run: |
+          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
```

Comment:
Can we try running this on the PR once, so we can test it locally? When I try testing locally, it can't find the image, I guess because it hasn't been pushed yet.

Reply:
As already commented https://github.com/PRQL/prql/pull/1893#issuecomment-1455409575, I will separate this part into a separate PR and merge it first so that we can try it out.